### PR TITLE
Icons: Added prometheus and traces icons

### PR DESCRIPTION
**Pull request by harisrozajac**
This PR adds prometheus and traces icons created by @diegoadams that will be used for Content Outline feature in Explore. 

Here's what the content outline will look like:
Screenshot 2023-08-24 at 9 53 42 AM

Also, here's more context on Content Outline in Explore: [link](https://github.com/orgs/grafana/projects/111/views/6?filterQuery=+-status%3AArchived+-label%3Atype%2Fepic+-type%3AProject%2CEpic%2CIdea+haris&pane=issue&itemId=24338048)

Fixes: https://github.com/grafana/grafana/issues/73691

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

File: packages/grafana-ui/src/components/Icon/cached.json [hunk 1]
```diff
@@ -160,6 +160,8 @@
   "custom/gf-logs",
   "custom/gf-movepane-left",
   "custom/gf-movepane-right",
+  "custom/gf-prometheus",
```
Comment (academo) [inline packages/grafana-ui/src/components/Icon/cached.json:163]: unless these icons will be displayed first thing when the pa

Comment (academo) [inline packages/grafana-ui/src/components/Icon/cached.json:163]: There's no need to add these icons to the js bundle if they will only exist behind a feature flag for now. It'll increase the size of the JS bundle but not be used.

Could you remove them from this list?

Comment (harisrozajac) [reply packages/grafana-ui/src/components/Icon/cached.json:163]: @academo thank you for your feedback. sure, I can do that.

**Comment by diegoadams** [comment]
@harisrozajac , this looks great🙂 I justed want to check the positioning of the Outline. I noticed it is on the right instead of the left side. Perhaps I missed some of the decision updates while I was away.

**Review by academo — APPROVED** [review]

**Pull request merged**